tests/kernel/common/testcase.yaml

```diff
@@ -29,6 +29,7 @@ tests:
       - CONFIG_CBPRINTF_FULL_INTEGRAL=y
   kernel.common.picolibc:
     filter: CONFIG_PICOLIBC_SUPPORTED
+    toolchain_exclude: arcmwdt
```

Comment:
should modify `CONFIG_PICOLIBC_SUPPORTED` to be `n` when `arcmwdt` is selected.

Reply:
What is needed to make picolibc usable on arc mwdt toolchain?

Reply:
something like this?

[0001-libc-picolibc-Picolibc-is-not-supported-using-the-AR.txt](https://github.com/zephyrproject-rtos/zephyr/files/9303287/0001-libc-picolibc-Picolibc-is-not-supported-using-the-AR.txt)

Reply:
Thanks @keith-packard !

Changed to proposed implementation.